doc/irmin/index.mld

```diff
@@ -0,0 +1,32 @@
+{0 Irmin}
+
+[Irmin] is a distributed database built on the same principles as {{:https://git-scm.com} Git}. It can be used as an OCaml library that provides persistent stores with built-in snapshot, branching and reverting mechanisms.
+
+{e Release %%VERSION%% - %%HOMEPAGE%%}
+
+{1 Irmin API}
+
+The [irmin] package defines the public API of [Irmin]. The entry point of the library is the module {!Irmin}.
+
+{1 Backends}
+
+Irmin is designed to use a large variety of backends. Various backends are provided:
+
+- {!module:Irmin_mem}: An in-memory store provided in the [irmin] package (this package).
+- {!Irmin_git}: A backend that exposes a bi-directional bridge to a Git repository proivded by the [irmin-git] package.
```

Comment:
I tested locally, and I wish there were a way to have the `{!Irmin_git}` (and similar) reference resolve. Is there any trick/hack to having it work that you know of?

Reply:
I wish for that as well! :) Unfortunately, I don't know of any trick or hack to make it work.

Maybe the new OCaml site can do it? It seems to be able to resolve the reference to `Repr` and jump to the correct package: https://ocaml.org/p/irmin/3.4.3/doc/Irmin/index.html

Reply:
Ah, nice! :crossed_fingers: this works magically with the next release.

Reply:
Sadly not. These links will only resolve if there's a actual dependency on package in which the referred module exists. Since `irmin-pack` doesn't depend upon `irmin-git` the module won't be there and odoc won't be able to resolve it. The only currently possible way to do it is an http link to the docs site itself, so: `{{: https://ocaml.org/p/irmin-git/}the irmin-git package}`

Reply:
Thanks for the clarification @jonludlam :pray: